### PR TITLE
Add company name to list of recent interactions on home page

### PR DESCRIPTION
**Pull request by reupen**
Port of https://github.com/uktrade/data-hub-prototype/pull/259

### CODE REVIEW

File: src/services/dashboardservice.js [hunk 1]
```diff
@@ -23,10 +23,15 @@ function mapContacts (contacts) {
 function mapInteractions (interactions) {
   if (interactions && (typeof interactions.map) === 'function') {
     return interactions.map((interaction) => {
+      const company = typeof interaction.company === 'object' ? interaction.company : {}
       return {
-        url: `/interaction/${interaction.id}`,
+        url: `/interaction/${interaction.id}/details`,
         id: interaction.id,
-        subject: interaction.subject
+        subject: interaction.subject,
```
Comment (ztolley) [inline src/services/dashboardservice.js]: Outside the scope of the ticket but we should limit the length on subject as some people put in EPIC subjects.

File: src/views/index.html [hunk 2]
```diff
@@ -19,6 +19,10 @@ <h2 class="dashboard-section-title">Interactions</h2>
           <ol id="interactions-list" class="dashboard-list">
           {% for interaction in interactions %}
             <li>
+              {% if interaction.company.name %}
```
Comment (ztolley) [inline src/views/index.html]: May want to check company property exists too. It's a required field and if the API is working it's not a problem but just a thought.

File: test/services/dashboardservice.test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  it('returns correct homepage data', () => {
+    return dashboardService.getHomepageData('token')
+      .then((data) => {
+        expect(data).to.eql({
+          interactions: [{
+            id: 'int-id',
+            subject: 'int-subject',
+            url: '/interaction/int-id/details',
+            company: {
+              'url': '/company/company_company/comp-id/details',
```
Comment (ztolley) [inline test/services/dashboardservice.test.js]: inconsistent syntax for labels. Sometimes with ', sometimes not.

Comment (reupen) [inline test/services/dashboardservice.test.js]: Fixed.

File: src/views/index.html (hunk 2 above)
Comment (reupen) [inline src/views/index.html]: It's protected against in dashboardservice.js. Will add a test though.

**Pull request merged**